apis/python/HISTORY.md

```diff
@@ -25,6 +25,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 ### Security
 
+- \[[#4031](https://github.com/single-cell-data/TileDB-SOMA/pull/4031)\] [python] Encode URI when generating from key
```

Comment:
could you expand this so it summarizes the change?  I'm thinking something like:

Storage paths generated from a collection keys will be URL-escaped when they contain characters outside of safe characters (defined as `a-zA-Z0-9-_.()^!@+={}~'`). In addition, the storage paths `..` and `.` are prohibited.